packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/MutableRealmInt.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.types
+
+import io.realm.kotlin.internal.UnmanagedMutableRealmInt
+
+/**
+ * A `MutableRealmInt` is a mutable, [Long]-like, numeric quantity. It behaves almost exactly as a
+ * reference to a [Long].
+ *
+ * `MutableRealmInt`s are most interesting as members of a managed [RealmObject] object. When
+ * managed, the [increment] and [decrement] operators implement a
+ * [conflict-free replicated data type](https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type):
+ * simultaneous increments and decrements from multiple distributed clients will be aggregated
+ * correctly. For instance, if the value of a `counter` field for the object representing a `User`
+ * named Fred is currently 0, then the following code, executed on two different devices,
+ * simultaneously, even if connected by only a slow, unreliable network, will **always** cause the
+ * value of `counter` to eventually converge on the value 2.
+ * ```
+ * realm.query<Sample>("name = $0", "Fred")
+ *      .first()
+ *      .find()!!
+ *      .counter
+ *      .increment(1)
+ * ```
+ * Note that the [set] operator must be used with extreme care. It will quash the effects of any
+ * prior calls to [increment] or [decrement]. Although the value of a `MutableRealmInt` will always
+ * converge across devices, the specific value on which it converges will depend on the actual order
+ * in which operations took place. Mixing [set] with [increment] and/or [decrement] is, therefore,
+ * not advised, unless fuzzy counting is acceptable.
+ *
+ * `MutableRealmInt`s **may not be primary keys**.
```

Comment:
```suggestion
 * `MutableRealmInt`s **cannot be primary keys**.
```

Reply:
Is this tested anywhere? At least I didn't seem to be able to find it

Reply:
It is, the compiler `PrimaryKeyTests` do it (https://github.com/realm/realm-kotlin/blob/el/mutable-realm-integer/test/base/src/jvmTest/kotlin/io/realm/kotlin/test/compiler/PrimaryKeyTests.kt#L53). Also in the library `PrimaryKeyTests` we check the number of supported types matches the expected types written in the test (https://github.com/realm/realm-kotlin/blob/el/mutable-realm-integer/test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/PrimaryKeyTests.kt#L156)